src/shared/ptyfwd.c

```diff
@@ -509,7 +509,8 @@ static int pty_forward_ansi_process(PTYForward *f, size_t offset) {
                         } else {
                                 /* Otherwise, the OSC sequence is over */
 
-                                if (c == '\x07') {
+                                bool is_string_terminator = (c == '\x1b' && (i + 1) < f->out_buffer_full && f->out_buffer[i + 1] == '\\');
```

Comment:
hmm, this is not ok, we can't really look ahead. While usually we'll have the whole sequence loaded we might not have that in some corner cases.

I'd probably just dumb this down, and just accept \x1b as end of sequence here (and not bother checking for the final \\). Just add a comment that we shortcut things here, and don't bother with checking for the 2nd part of the ST